### PR TITLE
Fix BlockSystems iterator

### PR DESCRIPTION
**Pull request by Pi-tree**
I fixed some problems with the iterator for BlockSystems.

### CODE REVIEW

**Comment by thofma** [comment]
Great. Do you think you could a small test to https://github.com/oscar-system/Oscar.jl/blob/master/test/Examples/galois-test.jl?

**Comment by Pi-tree** [comment]
Ok, I will do this

File: examples/GaloisGrp.jl [hunk 1]
```diff
@@ -865,6 +865,9 @@ function Base.iterate(B::BlockSystems)
 end
 
 function Base.iterate(B::BlockSystems, st::Array{Array{Int, 1}})
+if B.l==1||B.l==B.n
+  return nothing
+else
```
Comment (fingolfin) [inline examples/GaloisGrp.jl]: Better to do this: 
```suggestion
  if B.l==1||B.l==B.n
    return nothing
  end
```

File: examples/GaloisGrp.jl [hunk 2]
```diff
@@ -907,6 +913,7 @@ function Base.iterate(B::BlockSystems, st::Array{Array{Int, 1}})
     end
   end
 end
+end
```
Comment (fingolfin) [inline examples/GaloisGrp.jl]:
```suggestion
```

**Pull request merged**